docker/server/Dockerfile

```diff
@@ -3,23 +3,31 @@ SHELL ["/bin/bash", "-c"]
 ENV DEBIAN_FRONTEND=noninteractive
 ARG KERNEL_VERSION="5.8.0-53-generic"
 
-RUN apt-get update
-RUN apt-get install -y linux-image-${KERNEL_VERSION}
-RUN apt-get install -y wget git python3 python3-pip
+RUN \
+    apt-get update && \
+    apt-get install -y linux-image-${KERNEL_VERSION} && \
+    apt-get install -y wget git python3 python3-pip
 
-RUN mkdir /ota-server
 WORKDIR /ota-server
 
+# prepare environment
+RUN \
+    wget http://cdimage.ubuntu.com/ubuntu-base/releases/20.04/release/ubuntu-base-20.04.1-base-amd64.tar.gz && \
+    mkdir -p data && \
+    tar zxf ubuntu-base-20.04.1-base-amd64.tar.gz -C data && \
+    cp /boot/vmlinuz-${KERNEL_VERSION} /boot/initrd.img-${KERNEL_VERSION} /boot/config-${KERNEL_VERSION} /boot/System.map-${KERNEL_VERSION} data/boot
 
-RUN wget http://cdimage.ubuntu.com/ubuntu-base/releases/20.04/release/ubuntu-base-20.04.1-base-amd64.tar.gz
-RUN mkdir -p data
-RUN tar zxf ubuntu-base-20.04.1-base-amd64.tar.gz -C data
-RUN cp /boot/vmlinuz-${KERNEL_VERSION} /boot/initrd.img-${KERNEL_VERSION} /boot/config-${KERNEL_VERSION} /boot/System.map-${KERNEL_VERSION} data/boot
-RUN git clone https://github.com/tier4/ota-metadata
-RUN python3 -m pip install -r ota-metadata/metadata/ota_metadata/requirements.txt
-RUN python3 ota-metadata/metadata/ota_metadata/metadata_gen.py --target-dir data --ignore-file ota-metadata/metadata/ignore.txt 
-RUN ./ota-metadata/metadata/key-gen.sh
-RUN python3 ota-metadata/metadata/ota_metadata/metadata_sign.py --sign-key privatekey.pem --cert-file certificate.pem --persistent-file ota-metadata/metadata/persistents.txt --rootfs-directory data
-RUN cp ota-metadata/metadata/persistents.txt .
+# create ota-baseimage
+RUN \
+    git clone https://github.com/tier4/ota-metadata && \
```

Comment:
By this modification, docker cache cannot be used when `RUN` is failed.
I'd like to know this merit.

Reply:
>  to reduce unnecessary intermediate layers

I think for the Dockerfile developer, the layers are necessary to develop. When I try to add a new `RUN` line, I think I want to use layer cache to debug faster.

Reply:
> apt update/upgrade

is OK for apt cache problem.

> break up the create ota-image step into more parts

Yes. I'd like to prioritize development productivity compare to generating intermediate layers.

Reply:
in commit 40843f2, the second step breaks up into prepare metadata build environment, and creating ota-baseimage

Reply:
I think `RUN` for each command is enough. It means the original Dockerfile is enough (other than apt-get update&&apt-get update).
What do you think?

Reply:
OK I think one line per RUN is OK start from generating the metadata, since the previous steps are independent from our code.

Please check the commit bdf4d85.